src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -378,6 +421,84 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         }
     }
 
+    // This is called via the "install python" command in the kernel picker in the case where
+    // we have the python extension installed, but 0 valid python kernels / interpreters found
+    // just pop up a dialog box to prompt the user on how to install python
+    // Unlike installing the python extension we don't expect in progress executions to be handled
+    // when this command is installed, user will have to manually install python and rerun the cell
+    private async installPythonViaKernelPicker(): Promise<void> {
+        sendTelemetryEvent(Telemetry.PythonNotInstalled, undefined, { action: 'displayed' });
+        const selection = await this.appShell.showErrorMessage(
+            DataScience.pythonNotInstalledNonMarkdown(),
+            { modal: true },
+            Common.install()
+        );
+
+        if (selection === Common.install()) {
+            sendTelemetryEvent(Telemetry.PythonNotInstalled, undefined, { action: 'download' });
+            // Direct the user to download from python.org
+            this.appShell.openUrl('https://www.python.org/downloads');
+        } else {
+            sendTelemetryEvent(Telemetry.PythonNotInstalled, undefined, { action: 'dismissed' });
+        }
+    }
+
+    // Called when we select the command to install the python extension via the kernel picker
+    // If new controllers are added before this fully resolves any in progress executions will be
+    // passed on, so we can trigger with the run button, install, get a controller and not have to
+    // click run again
+    private async installPythonExtensionViaKernelPicker(): Promise<void> {
```

Comment:
The notebookControllerManager is very complex right now.
I think this can go into a seperate class, all we're doing here is just handling the command, 
I think we can move the handler along with the command registration into a separate class.